rad/prelude/io.rad

```diff
@@ -189,6 +192,23 @@ process exited normally and `[:error n]` otherwise. Example: `(process!
   [ (process-with-stdout-stderr-exitcode! "false" [] "") ==> [[] [] [:error 1]] ]
 )
 
+(def process-with-stdout-strict!
+  "Like `process-with-stdout!`, but prints an error message and exits if the command fails."
+  (fn [command args to-write]
+    (match (process-with-stdout-stderr-exitcode! command args to-write)
+      ['stdout _ :ok] stdout
+      ['stdout 'stderr [:error 'n]]
+        (do
+          (def command-line (strings/intercalate " " (cons command args)))
+          (put-str! (string-append "error: Command \"" command-line "\" exited with status code " (show n)))
+          (map (fn [l] (put-str! (string-append "  stderr: " l))) stderr)
```

Comment:
As this is a user facing error, it should be moved to `error-messages.rad`. It might also make sense to pass part of the error message to the function so that we can print specific instructions for the user, but it depends on where it is/will be used.

Reply:
I extracted the error message but I’m not sure about the second part. Currently the function is only used in one place where it does not make sense to pass an error message. If we need that functionality in the future we can add it.